### ISSUE TITLE
Offensive/racist words in words.txt

### ISSUE THREAD

**Issue by yanlingpsu** [AllenDowney/ThinkPython#5]
The words.txt file contains racial slurs and other inappropriate words (for example, the n word). It should be updated with the offensive content removed.

**Comment by AllenDowney** [comment]
I've removed words I thought most likely to be problematic, especially in the context of a class.  I am open to suggestions for additional removals.

**Comment by yanlingpsu** [comment]
Hi Allen,

Thank you for your fast response.

I had to respond to a complaint from my students regarding this issue and
that is why I sent the email to you earlier. However, as I think more about
what this file really is, it is a dictionary. And in fact I used the list
in one of the assignments to do spell-checking of online pulled txt files
like Shakespeare, constitution and other documents. Removing proper English
words from the dictionary will render these tasks impossible, or it will
actually pull out the racial slurs in Shakespeare as misspelled words. I am
not sure what is the best approach to this problem. The whole world is
becoming a lot more sensitive and this year made things even more difficult.

Again, thank you so much for your fast response. And our students and I
really appreciated using your textbook Think Python this semester. It was
so well written and a not too heavy read that gets straight to the point.

Yanling Wang

**Comment by AllenDowney** [comment]
I use the word list for a few different exercises, but the first one is searching for words that have various properties, like "three double letters" and things like that.  So words from the word list are going to be selected without anyone looking for them specifically.  And there are certainly words that would not be welcome in that context.  I removed the ones that I thought would be most jarring if they appeared in the middle of a word game.

In a later exercise, we identify words in a document that don't appear in the dictionary (like a spell checker).  So if the document contains the words I removed, it will cause "false positives" for misspelled words.  I think I can live with that, to avoid the first issue.
